pkg/connector/handlematrix.go

```diff
@@ -472,3 +469,67 @@ func (s *SignalClient) HandleMatrixMembership(ctx context.Context, msg *bridgev2
 	msg.Portal.Metadata.(*signalid.PortalMetadata).Revision = revision
 	return true, nil
 }
+
+func plToRole(pl int) signalmeow.GroupMemberRole {
+	if pl >= moderatorPL {
+		return signalmeow.GroupMember_ADMINISTRATOR
+	} else {
+		return signalmeow.GroupMember_DEFAULT
+	}
+}
+
+func hasAdminChanged(plc *bridgev2.SinglePowerLevelChange) bool {
+	if plc == nil {
+		return false
+	}
+	return (plc.NewLevel < moderatorPL) != (plc.OrigLevel < moderatorPL)
+}
+
+func (s *SignalClient) HandleMatrixPowerLevels(ctx context.Context, msg *bridgev2.MatrixPowerLevelChange) (bool, error) {
+	if msg.Portal.RoomType == database.RoomTypeDM {
+		return false, nil
+	}
+	log := zerolog.Ctx(ctx)
+	gc := &signalmeow.GroupChange{}
+	for _, plc := range msg.Users {
+		if !hasAdminChanged(&plc.SinglePowerLevelChange) {
+			continue
+		}
+		aci, err := signalid.ParseGhostOrUserLoginID(plc.Target)
+		if err != nil {
+			log.Err(err).Msg("Couldn't parse user id")
+		}
+		gc.ModifyMemberRoles = append(gc.ModifyMemberRoles, &signalmeow.RoleMember{
+			ACI:  aci,
+			Role: plToRole(plc.NewLevel),
+		})
+	}
+	if hasAdminChanged(msg.EventsDefault) {
+		announcementsOnly := msg.EventsDefault.NewLevel >= moderatorPL
+		gc.ModifyAnnouncementsOnly = &announcementsOnly
+	}
+	if hasAdminChanged(msg.StateDefault) {
+		attributesAccess := signalmeow.AccessControl_MEMBER
+		if msg.StateDefault.NewLevel >= moderatorPL {
+			attributesAccess = signalmeow.AccessControl_ADMINISTRATOR
+		}
```

Comment:
This could probably be a `plToAccessControl` helper function since it's used twice